app/src/main/java/org/wikipedia/richtext/CustomHtmlParser.kt

```diff
@@ -0,0 +1,147 @@
+        if (tagStatus.isNotEmpty() && !tagStatus.removeLast()) {
+            wrapped?.endElement(uri, localName, qName)
+        }
+        handler.handleTag(false, localName, text, null)
+    }
+
+    override fun setDocumentLocator(locator: Locator?) {
+        wrapped?.setDocumentLocator(locator)
+    }
+
+    override fun startDocument() {
+        wrapped?.startDocument()
+    }
+
+    override fun endDocument() {
+        wrapped?.endDocument()
+    }
+
+    override fun startPrefixMapping(prefix: String?, uri: String?) {
+        wrapped?.startPrefixMapping(prefix, uri)
+    }
+
+    override fun endPrefixMapping(prefix: String?) {
+        wrapped?.endPrefixMapping(prefix)
+    }
+
+    override fun characters(ch: CharArray?, start: Int, length: Int) {
+        wrapped?.characters(ch, start, length)
+    }
+
+    override fun ignorableWhitespace(ch: CharArray?, start: Int, length: Int) {
+        wrapped?.ignorableWhitespace(ch, start, length)
+    }
+
+    override fun processingInstruction(target: String?, data: String?) {
+        wrapped?.processingInstruction(target, data)
+    }
+
+    override fun skippedEntity(name: String?) {
+        wrapped?.skippedEntity(name)
+    }
+
+    class CustomTagHandler(private val view: TextView?) : TagHandler {
+        private var lastAClass = ""
+
+        override fun handleTag(opening: Boolean, tag: String?, output: Editable?, attributes: Attributes?): Boolean {
+            if (tag == "img") {
+                return true
+            } else if (tag == "a") {
+                if (opening) {
+                    lastAClass = getValue(attributes, "class").orEmpty()
+                } else if (output != null && output.isNotEmpty()) {
+                    val spans = output.getSpans<URLSpan>(output.length - 1)
+                    if (spans.isNotEmpty()) {
+                        val span = spans.last()
+                        val start = output.getSpanStart(span)
+                        val end = output.getSpanEnd(span)
+                        output.removeSpan(span)
+                        val color = if (lastAClass == "new" && view != null) ResourceUtil.getThemedColor(view.context, R.attr.colorError) else -1
+                        output.setSpan(URLSpanNoUnderline(span.url, color), start, end, 0)
+                    }
+                }
+            }
+            return false
+        }
+    }
+
+    companion object {
+
+        fun fromHtml(html: String?, view: TextView? = null): Spanned {
+            var sourceStr = html.orEmpty()
+
+            if ("<" !in sourceStr && "&" !in sourceStr) {
+                // If the string doesn't contain any hints of HTML entities, then skip the expensive
+                // processing that fromHtml() performs.
+                return sourceStr.toSpanned()
+            }
+
+            sourceStr = sourceStr.replace("&#8206;", "\u200E")
+                .replace("&#8207;", "\u200F")
+                .replace("&amp;", "&")
```

Comment:
Would it be possible to add comments for those encoded strings `&#8206;`, `&#8207;` and `&amp`?